src/main/code.ts

```diff
@@ -5,125 +5,445 @@
+		for (var i = 0; i < selectedFrame.children.length; i++) {
+			var column = selectedFrame.children[i] as FrameNode
+			if ("children" in column) {
+				for (var j = 0; j < column.children.length; j++) {
+					if (column.children[i].type != "INSTANCE") {
+						resizable = false
+						j = column.children.length
+						i = selectedFrame.children.length
+					}
+				}
+			} else {
+				figma.closePlugin("🛑 There is a column with no cells");
 			}
 
+		}
+	}
+
+	// Getting the numbers of columns of the selected table and comparing it to the data
+	var columnLength = selectedFrame.children.length
+	if (columnLength != columnLengthData) {
+		if (resizable) {
+			addOrRemoveColumns(selectedFrame, columnLength, columnLengthData)
+			columnLength = columnLengthData
+		} else if (isColumn) {
+			// Column selected
+			for (var i = 0; i < selectedFrame.children.length; i++) {
+				var textNode = getFirstChildOfTypeText(selectedFrame.children[i] as SceneNode)
+				if (textNode != null) {
+					textNodes.push(textNode)
+				}
+			}
+
+			if (columnLengthData == 1 && rowLengthData == textNodes.length) {
+				insertData(textNodes, cellsData, right_align_numbers)
+			} else {
+				figma.closePlugin("🛑 This seems to be a column with a different cell count than given by your data");
+			}
+			return
 		} else {
-		  getFirstChildOfTypeText(newNode).textAlignHorizontal = 'LEFT' // fails on "'" character
+			figma.closePlugin("🛑 The column count from your selected node is different than your given data column count");
+			return
 		}
-		
-		if (wrapInAutoLayout) {
-		  columnAutoLayouts[j].appendChild(newNode)
-		  columnAutoLayouts[j].counterAxisSizingMode = 'AUTO'
+	}
+
+	// Comparing the row length of the selected table with the data
+	var rowLength = getTextNodes(selectedFrame, textNodes, columnLength)
+	if (rowLength != rowLengthData) {
+		if (resizable) {
+			updateRowCount(selectedFrame, columnLength, rowLength, rowLengthData)
+			textNodes = []
+			rowLength = getTextNodes(selectedFrame, textNodes, columnLength)
 		} else {
-		  newNode.x = original.x + original.width * (j + 1)
-		  newNode.y = original.y + original.height * (i + 1)
+			figma.closePlugin("🛑 Your given data has not the same row count as your selected table");
+			return
 		}
+	}
 
-		if (i == 0) {
-			headerCells.push(newNode)
+	// Copy text from data cell into table
+	insertData(textNodes, cellsData, right_align_numbers)
+}
+
+/**
+ * This function determines the  row count of the table looking into the cell amount of each column.
+ * At the same time it adds all text frames that it finds on the way into the given array "textFrames".
+ * It only works after knowing the column count because cells are inside columns which are inside the table.
+ *
+ * @param selectedFrame - The selected parent node as a Frame
+ * @param textFrames - An empty array to fill all found text frames into
+ * @param columnLength - The column count of the table (was calculated before this method)
```

Comment:
```suggestion
 * @param columnCount - The column count of the table (must be calculated before calling this method)
```

Reply:
`ColumnLength` would rather denote the number of rows _within_ a column.

Reply:
I will change that inside the method as well as at the "@param" spot.